src/views/ExportSuccess.vue

```diff
@@ -83,8 +83,7 @@ export default class ExportSuccess extends Vue {
 
             // Recreate original URL with original query parameters
             const query = { rpcId: staticStore.rpcState!.id.toString() };
-            setTimeout(
-                () => this.$router.push({ name: RequestType.MIGRATE, query }), StatusScreen.SUCCESS_REDIRECT_DELAY);
+            setTimeout(() => this.$rpc.routerPush(RequestType.MIGRATE, query), StatusScreen.SUCCESS_REDIRECT_DELAY);
```

Comment:
Looks to me as if we might need to take action to prevent us using `this.$router.push()` and `this.$router.replace()` as they expose a behaviour which is no longer desirable?

Reply:
What would you suggest, removing the `$router` from our vue instances or overwriting `router.push` to print a warning?

Reply:
I think removing `this.$router` is not recommended, because it's set internally by vue-router when the plugin is activated, and likely also used internally by the plugin.

I think it makes sense to add a navigation guard in dev mode (https://router.vuejs.org/guide/advanced/navigation-guards.html#navigation-guards), which checks that the navigation was done with the correct method and warn (or even block) in dev mode.

Reply:
Could we remove the `router_replace/push` methods from RpcApi in favour of the navigation guards? 
If not this still seems to me like the preferred solution as well.

Reply:
Good idea :+1:
I opened an issue for this: https://github.com/nimiq/hub/issues/371